app/src/main/java/com/example/playlistmaker/playlist/ui/fragments/PlaylistFragment.kt

```diff
@@ -0,0 +1,284 @@
+                    binding.playlistDescriptionTextView.text =
+                        state.playlistModel.playlistDescription
+                    binding.tracksTimeTextView.text = resources.getQuantityString(
+                        R.plurals.minute_plural, minutesCount, minutesCount
+                    )
+                    binding.tracksCountTextView.text = resources.getQuantityString(
+                        R.plurals.tracks_plural, tracksCount, tracksCount
+                    )
+
+                    //init menu playlist view
+                    Glide.with(this).load(state.playlistModel.playlistCoverImage)
+                        .placeholder(R.drawable.no_album).centerCrop()
+                        .transform(RoundedCorners(resources.getDimensionPixelSize(R.dimen.album_cover_corner_radius)))
+                        .into(binding.bottomSheetMenu.playlistView.playlistCoverImageView)
+
+                    binding.bottomSheetMenu.playlistView.playlistNameTextView.text =
+                        state.playlistModel.playlistName
+                    binding.bottomSheetMenu.playlistView.trackCountTextView.text =
+                        resources.getQuantityString(
+                            R.plurals.tracks_plural, tracksCount, tracksCount
+                        )
+
+                    //update recycle view
+                    tracksAdapter?.updateAdapter(state.playlistModel.tracks)
+                }
+
+            }
+
+            PlaylistState.CloseScreen -> findNavController().popBackStack()
+        }
+    }
+
+    private fun renderSharePlaylistState(state: SharePlaylistState) {
+        when (state) {
+            SharePlaylistState.Error -> showToast(getString(R.string.share_playlist_error))
+            SharePlaylistState.Success -> viewModel.sharePlaylist()
+        }
+    }
+
+    private fun showToast(message: String) {
+        Toast.makeText(context, message, Toast.LENGTH_LONG).show()
+    }
+
+    private fun deleteTrackConfirmDialogRequest(it: Track) {
+        confirmDialog?.setTitle(R.string.delete_track_title)?.setMessage(R.string.delete_track)
+            ?.setNegativeButton(R.string.cancel) { _, _ ->
+            }?.setPositiveButton(R.string.yes_delete) { _, _ ->
+                viewModel.deleteTrack(it)
+            }
+        confirmDialog?.show()
+    }
+
+    private fun deletePlaylistConfirmDialogRequest() {
+
+        confirmDialog?.setTitle(R.string.delete_playlist_title)
+            ?.setMessage(resources.getString(R.string.delete_playlist, viewModel.getPlaylistName()))
+            ?.setNegativeButton(R.string.no) { _, _ ->
+            }?.setPositiveButton(R.string.yes) { _, _ ->
+                viewModel.deletePlaylist()
+            }
+        confirmDialog?.show()
+    }
+
+    private fun getTotalTracksTime(tracks: List<Track>): Int {
+        var durationSum = 0L
+        tracks.forEach {
+            durationSum += it.trackTimeMillis
+        }
```

Comment:
🍏 Можно использовать специальную функцию Kotlin - fold. https://kotlinlang.org/docs/collection-aggregate.html#fold-and-reduce
```suggestion
        val durationSum = tracks.fold(0L) { sum, element -> sum + element.trackTimeMillis }
```